src/components/Header.tsx

```diff
@@ -0,0 +1,63 @@
+import React, {
+  FC,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  addTodo: (todo: Todo) => void;
+};
+
+export const Header: FC<Props> = ({ addTodo }) => {
+  const [inputValue, setInputValue] = useState('');
+  const newTodoField = useRef<HTMLInputElement>(null);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setInputValue(value);
+  };
+
+  const handleFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (inputValue.length === 0) {
+      return;
+    }
+
+    const newTodo = {
+      id: +new Date(),
```

Comment:
Using the current date timestamp as an id may not guarantee uniqueness. Consider using a library like uuid to generate unique ids.